### PR TITLE
Fix recipe book not being empty when using a multi-server configuration, with servers which are pre-1.12.

### PR DESCRIPTION
**Pull request by valaphee**
# Description
When using a multi-server configuration, with servers which are pre-1.12, switching from 1.12 to pre-1.12 causes the recipe book to not clear.

# Testing
Can be tested with a BungeeCord and two servers, one is pre-1.12 and the other one 1.12 or later. When switching from 1.12 to 1.11, the recipe book is still shown, even when unusable and containing invalid recipes.

### CODE REVIEW

File: common/src/main/java/com/viaversion/viaversion/protocols/protocol1_12to1_11_1/Protocol1_12To1_11_1.java [hunk 1]
```diff
@@ -167,6 +169,13 @@ public void registerMap() {
                     ClientWorld clientChunks = wrapper.user().get(ClientWorld.class);
                     int dimensionId = wrapper.get(Type.INT, 1);
                     clientChunks.setEnvironment(dimensionId);
+
+                    wrapper.create(ClientboundPackets1_13.DECLARE_RECIPES, new PacketHandler() { // Reset recipes
+                        @Override
+                        public void handle(PacketWrapper wrapper) throws Exception {
+                            wrapper.write(Type.VAR_INT, 0);
+                        }
+                    }).scheduleSend(Protocol1_13To1_12_2.class);
```
Comment (kennytv) [inline common/src/main/java/com/viaversion/viaversion/protocols/protocol1_12to1_11_1/Protocol1_12To1_11_1.java:178]: Please make the handler a lambda

File: common/src/main/java/com/viaversion/viaversion/protocols/protocol1_12to1_11_1/Protocol1_12To1_11_1.java [hunk 2]
```diff
@@ -167,6 +169,10 @@ public void registerMap() {
                     ClientWorld clientChunks = wrapper.user().get(ClientWorld.class);
                     int dimensionId = wrapper.get(Type.INT, 1);
                     clientChunks.setEnvironment(dimensionId);
+
+                    // Reset recipes
+                    wrapper.create(ClientboundPackets1_13.DECLARE_RECIPES, packetWrapper -> packetWrapper.write(Type.VAR_INT, 0))
+                            .scheduleSend(Protocol1_13To1_12_2.class);
```
Comment (kennytv) [inline common/src/main/java/com/viaversion/viaversion/protocols/protocol1_12to1_11_1/Protocol1_12To1_11_1.java:175]: Oh and this should be `Protocol1_12To1_11_1.class`, the default behavior of that method is to skip current pipeline (aka start from the target)

**Comment by kennytv** [comment]
What I missed: You need to make sure the client is a actually on 1.13+ before sending something through the 1.13 pipeline

**Comment by valaphee** [comment]
Ah thanks, wasn't sure if it just gets omitted

**Review by kennytv — APPROVED** [review]
Thanks!

**Pull request merged**